FairMQ Installation fails in setting dev

Dear developers,

I have another issue with the installation of FairSoft via SPACK.

```
Installing fairmq
==> Searching for binary cache of fairmq
==> Warning: No Spack mirrors are currently configured
==> No binary for fairmq found: installing from source
==> Warning: microarchitecture specific optimizations are not supported yet on mixed compiler toolchains [check clang@10.0.1-apple for further details]
==> Cloning git repository: https://github.com/FairRootGroup/FairMQ.git at commit 9f8a3553ba7e6c3ce5c9e378d77d3e16d7ce6daf
==> No checksum needed when fetching with git
==> Already staged spack-stage-fairmq-1.4.12-j6hrtuzhq7y7yoo3dmfpuhtuoyqmzpqh in /var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tstockmanns/spack-stage/spack-stage-fairmq-1.4.12-j6hrtuzhq7y7yoo3dmfpuhtuoyqmzpqh
==> No patches needed for fairmq
==> Building fairmq [CMakePackage]
==> Executing phase: 'cmake'
==> Executing phase: 'build'
==> Error: ProcessError: Command exited with status 1:
    'ninja' '-j4' '-v'

15 errors found in build log:
     308    In file included from /opt/local/include/boost/process.hpp:23:
     309    In file included from /opt/local/include/boost/process/async.hpp:36:
     310    In file included from /opt/local/include/boost/asio/streambuf.hpp:22:
     311    In file included from /opt/local/include/boost/asio/basic_streambuf.hpp:28:
     312    In file included from /opt/local/include/boost/asio/buffer.hpp:27:
     313    In file included from /opt/local/include/boost/asio/detail/string_view.hpp:23:
  >> 314    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/experimental/string_view:11:2: error: "<experimental/string_view> has been removed. Use
             <string_view> instead."
     315    #error "<experimental/string_view> has been removed. Use <string_view> instead."
     316     ^
     317    In file included from /var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tstockmanns/spack-stage/spack-stage-fairmq-1.4.12-j6hrtuzhq7y7yoo3dmfpuhtuoyqmzpqh/spack-src/test/device/_exceptio
            ns.cxx:12:
     318    In file included from /opt/local/include/boost/process.hpp:23:
     319    In file included from /opt/local/include/boost/process/async.hpp:36:
     320    In file included from /opt/local/include/boost/asio/streambuf.hpp:22:
     321    In file included from /opt/local/include/boost/asio/basic_streambuf.hpp:28:
     322    In file included from /opt/local/include/boost/asio/buffer.hpp:27:
  >> 323    /opt/local/include/boost/asio/detail/string_view.hpp:32:12: error: no member named 'experimental' in namespace 'std'
     324    using std::experimental::basic_string_view;
     325          ~~~~~^
  >> 326    /opt/local/include/boost/asio/detail/string_view.hpp:33:12: error: no member named 'experimental' in namespace 'std'
     327    using std::experimental::string_view;
     328          ~~~~~^
     329    In file included from /var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tstockmanns/spack-stage/spack-stage-fairmq-1.4.12-j6hrtuzhq7y7yoo3dmfpuhtuoyqmzpqh/spack-src/test/device/_exceptio
            ns.cxx:12:
     330    In file included from /opt/local/include/boost/process.hpp:23:
     331    In file included from /opt/local/include/boost/process/async.hpp:36:
     332    In file included from /opt/local/include/boost/asio/streambuf.hpp:22:
     333    In file included from /opt/local/include/boost/asio/basic_streambuf.hpp:28:
  >> 334    /opt/local/include/boost/asio/buffer.hpp:1489:5: error: no template named 'basic_string_view'; did you mean 'std::basic_string_view'?
     335        basic_string_view<Elem, Traits> data) BOOST_ASIO_NOEXCEPT
     336        ^
     337    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/string_view:194:28: note: 'std::basic_string_view' declared here
     338    class _LIBCPP_TEMPLATE_VIS basic_string_view {
     339                               ^
     340    In file included from /var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tstockmanns/spack-stage/spack-stage-fairmq-1.4.12-j6hrtuzhq7y7yoo3dmfpuhtuoyqmzpqh/spack-src/test/device/_exceptio
            ns.cxx:12:
     341    In file included from /opt/local/include/boost/process.hpp:23:
     342    In file included from /opt/local/include/boost/process/async.hpp:36:
     343    In file included from /opt/local/include/boost/asio/streambuf.hpp:22:
     344    In file included from /opt/local/include/boost/asio/basic_streambuf.hpp:28:
  >> 345    /opt/local/include/boost/asio/buffer.hpp:1510:5: error: no template named 'basic_string_view'; did you mean 'std::basic_string_view'?
     346        basic_string_view<Elem, Traits> data,
     347        ^
     348    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/string_view:194:28: note: 'std::basic_string_view' declared here
     349    class _LIBCPP_TEMPLATE_VIS basic_string_view {
     350                               ^
     351    5 errors generated.

     ...

     356    In file included from /opt/local/include/boost/process.hpp:23:
     357    In file included from /opt/local/include/boost/process/async.hpp:36:
     358    In file included from /opt/local/include/boost/asio/streambuf.hpp:22:
     359    In file included from /opt/local/include/boost/asio/basic_streambuf.hpp:28:
     360    In file included from /opt/local/include/boost/asio/buffer.hpp:27:
     361    In file included from /opt/local/include/boost/asio/detail/string_view.hpp:23:
  >> 362    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/experimental/string_view:11:2: error: "<experimental/string_view> has been removed. Use
             <string_view> instead."
     363    #error "<experimental/string_view> has been removed. Use <string_view> instead."
     364     ^
     365    In file included from /var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tstockmanns/spack-stage/spack-stage-fairmq-1.4.12-j6hrtuzhq7y7yoo3dmfpuhtuoyqmzpqh/spack-src/test/device/_signals.
            cxx:12:
     366    In file included from /opt/local/include/boost/process.hpp:23:
     367    In file included from /opt/local/include/boost/process/async.hpp:36:
     368    In file included from /opt/local/include/boost/asio/streambuf.hpp:22:
     369    In file included from /opt/local/include/boost/asio/basic_streambuf.hpp:28:
     370    In file included from /opt/local/include/boost/asio/buffer.hpp:27:
  >> 371    /opt/local/include/boost/asio/detail/string_view.hpp:32:12: error: no member named 'experimental' in namespace 'std'
     372    using std::experimental::basic_string_view;
     373          ~~~~~^
  >> 374    /opt/local/include/boost/asio/detail/string_view.hpp:33:12: error: no member named 'experimental' in namespace 'std'
     375    using std::experimental::string_view;
     376          ~~~~~^
     377    In file included from /var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tstockmanns/spack-stage/spack-stage-fairmq-1.4.12-j6hrtuzhq7y7yoo3dmfpuhtuoyqmzpqh/spack-src/test/device/_signals.
            cxx:12:
     378    In file included from /opt/local/include/boost/process.hpp:23:
     379    In file included from /opt/local/include/boost/process/async.hpp:36:
     380    In file included from /opt/local/include/boost/asio/streambuf.hpp:22:
     381    In file included from /opt/local/include/boost/asio/basic_streambuf.hpp:28:
  >> 382    /opt/local/include/boost/asio/buffer.hpp:1489:5: error: no template named 'basic_string_view'; did you mean 'std::basic_string_view'?
     383        basic_string_view<Elem, Traits> data) BOOST_ASIO_NOEXCEPT
     384        ^
     385    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/string_view:194:28: note: 'std::basic_string_view' declared here
     386    class _LIBCPP_TEMPLATE_VIS basic_string_view {
     387                               ^
     388    In file included from /var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tstockmanns/spack-stage/spack-stage-fairmq-1.4.12-j6hrtuzhq7y7yoo3dmfpuhtuoyqmzpqh/spack-src/test/device/_signals.
            cxx:12:
     389    In file included from /opt/local/include/boost/process.hpp:23:
     390    In file included from /opt/local/include/boost/process/async.hpp:36:
     391    In file included from /opt/local/include/boost/asio/streambuf.hpp:22:
     392    In file included from /opt/local/include/boost/asio/basic_streambuf.hpp:28:
  >> 393    /opt/local/include/boost/asio/buffer.hpp:1510:5: error: no template named 'basic_string_view'; did you mean 'std::basic_string_view'?
     394        basic_string_view<Elem, Traits> data,
     395        ^
     396    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/string_view:194:28: note: 'std::basic_string_view' declared here
     397    class _LIBCPP_TEMPLATE_VIS basic_string_view {
     398                               ^
     399    5 errors generated.

     ...

     404    In file included from /opt/local/include/boost/process.hpp:23:
     405    In file included from /opt/local/include/boost/process/async.hpp:36:
     406    In file included from /opt/local/include/boost/asio/streambuf.hpp:22:
     407    In file included from /opt/local/include/boost/asio/basic_streambuf.hpp:28:
     408    In file included from /opt/local/include/boost/asio/buffer.hpp:27:
     409    In file included from /opt/local/include/boost/asio/detail/string_view.hpp:23:
  >> 410    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/experimental/string_view:11:2: error: "<experimental/string_view> has been removed. Use
             <string_view> instead."
     411    #error "<experimental/string_view> has been removed. Use <string_view> instead."
     412     ^
     413    In file included from /var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tstockmanns/spack-stage/spack-stage-fairmq-1.4.12-j6hrtuzhq7y7yoo3dmfpuhtuoyqmzpqh/spack-src/test/device/_error_st
            ate.cxx:12:
     414    In file included from /opt/local/include/boost/process.hpp:23:
     415    In file included from /opt/local/include/boost/process/async.hpp:36:
     416    In file included from /opt/local/include/boost/asio/streambuf.hpp:22:
     417    In file included from /opt/local/include/boost/asio/basic_streambuf.hpp:28:
     418    In file included from /opt/local/include/boost/asio/buffer.hpp:27:
  >> 419    /opt/local/include/boost/asio/detail/string_view.hpp:32:12: error: no member named 'experimental' in namespace 'std'
     420    using std::experimental::basic_string_view;
     421          ~~~~~^
  >> 422    /opt/local/include/boost/asio/detail/string_view.hpp:33:12: error: no member named 'experimental' in namespace 'std'
     423    using std::experimental::string_view;
     424          ~~~~~^
     425    In file included from /var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tstockmanns/spack-stage/spack-stage-fairmq-1.4.12-j6hrtuzhq7y7yoo3dmfpuhtuoyqmzpqh/spack-src/test/device/_error_st
            ate.cxx:12:
     426    In file included from /opt/local/include/boost/process.hpp:23:
     427    In file included from /opt/local/include/boost/process/async.hpp:36:
     428    In file included from /opt/local/include/boost/asio/streambuf.hpp:22:
     429    In file included from /opt/local/include/boost/asio/basic_streambuf.hpp:28:
  >> 430    /opt/local/include/boost/asio/buffer.hpp:1489:5: error: no template named 'basic_string_view'; did you mean 'std::basic_string_view'?
     431        basic_string_view<Elem, Traits> data) BOOST_ASIO_NOEXCEPT
     432        ^
     433    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/string_view:194:28: note: 'std::basic_string_view' declared here
     434    class _LIBCPP_TEMPLATE_VIS basic_string_view {
     435                               ^
     436    In file included from /var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tstockmanns/spack-stage/spack-stage-fairmq-1.4.12-j6hrtuzhq7y7yoo3dmfpuhtuoyqmzpqh/spack-src/test/device/_error_st
            ate.cxx:12:
     437    In file included from /opt/local/include/boost/process.hpp:23:
     438    In file included from /opt/local/include/boost/process/async.hpp:36:
     439    In file included from /opt/local/include/boost/asio/streambuf.hpp:22:
     440    In file included from /opt/local/include/boost/asio/basic_streambuf.hpp:28:
  >> 441    /opt/local/include/boost/asio/buffer.hpp:1510:5: error: no template named 'basic_string_view'; did you mean 'std::basic_string_view'?
     442        basic_string_view<Elem, Traits> data,
     443        ^
     444    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/string_view:194:28: note: 'std::basic_string_view' declared here
     445    class _LIBCPP_TEMPLATE_VIS basic_string_view {
     446                               ^
     447    5 errors generated.
```

--- a/packages/fairmq/package.py
+++ b/packages/fairmq/package.py
@@ -45,6 +45,7 @@ class Fairmq(CMakePackage):
     conflicts('cxxstd=11', when='@1.4.11:')
 
     patch('fix_find_dds.patch', when='@1.4.0:1.4.4')
+    patch('use_bundled_gtest_149.patch', when='@1.4.9:1.4.16')
 
     depends_on('googletest@1.7:', when='@:1.4.8')
     depends_on('boost@1.64: +container+program_options+thread+system+filesystem+regex+date_time', when='@1.3')

--- a/packages/fairmq/use_bundled_gtest_149.patch
+++ b/packages/fairmq/use_bundled_gtest_149.patch
@@ -1,0 +1,32 @@
+commit b32e04db6051024ce0439128c89cff153281cdec
+Author: Dennis Klein <d.klein@gsi.de>
+Date:   Tue May 19 10:23:15 2020 +0200
+
+    Do not search external GTest by default
+    
+    Can be overridden by -DUSE_EXTERNAL_GTEST=ON.
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index e1279fd4..6661b2b5 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -51,6 +51,8 @@ fairmq_build_option(BUILD_DOCS              "Build FairMQ documentation."
+                                              DEFAULT OFF)
+ fairmq_build_option(FAST_BUILD              "Fast production build. Not recommended for development."
+                                              DEFAULT OFF)
++fairmq_build_option(USE_EXTERNAL_GTEST      "Do not use bundled GTest. Not recommended."
++                                             DEFAULT OFF)
+ ################################################################################
+ 
+ 
+@@ -160,7 +160,9 @@
+ endif()
+ 
+ if(BUILD_TESTING)
+-  find_package2(PRIVATE GTest VERSION 1.7.0)
++  if(USE_EXTERNAL_GTEST)
++    find_package2(PRIVATE GTest VERSION 1.7.0)
++  endif()
+   if(NOT GTest_FOUND)
+     build_bundled(GTest extern/googletest)
+     find_package2(PRIVATE GTest REQUIRED)

